### PR TITLE
VPS 173/manage groups download functionality

### PR DESCRIPTION
**Pull request by codecreator127**
## Describe the issue

Download button in Manage Groups Page doesn't work

### Backend
- Implemented a ```getGroupByScenarioId``` function that fetches groups by scenarioId (under ```/db/daos/groupDao```).
- Made a new API endpoint at ```/api/group/scenario/:scenarioId``` that fetches groups by scenarioId using the getGroupByScenarioId function.

example usage of new API end point (this is fetchGroups function detailed below):
image

### Frontend
- Added a fetchGroups function to the Manage Groups Page (```src/containers/pages/ManageGroups```), which uses the new API point ```/api/group/scenario/:scenarioId``` to fetch that scenario's groups
- Refactored download function to use fetched data instead of dummy data

## Risk

May have merge conflicts

Download functionality shouldn't introduce any new problems, as this PR doesn't change any existing code (only adds new functionality)

## Definition of Done

- [ ] Code peer-reviewed
- [ ] Unit tests written and passing
- [ ] Integration tests written and passing
- [ ] Continuous Integration build passing
- [ ] Acceptance criteria met

## Reviewed By

Who reviewed your PR - for commit history once merged

### CODE REVIEW

File: frontend/src/containers/pages/ManageGroups/ManageGroupsPage.jsx [hunk 1]
```diff
@@ -56,9 +72,38 @@ export default function ManageGroupsPage() {
...
-  function download() {
-    console.log("downloading current group config as .CSV");
-  }
+  const convertToCSV = (data) => {
+    const headers = "email,first name,last name,group number,playable link\n";
+    let csv = headers;
+
+    data.forEach((row) => {
+      const entries = Object.entries(row);
+      entries.forEach(([, value]) => {
```
Comment (wjin-lee) [inline frontend/src/containers/pages/ManageGroups/ManageGroupsPage.jsx:81]: Note that the response of the endpoint you created seems to be

```json
[
    {
        "users": [
            {
                "email": "abc@gmail.com",
                "name": "abc aba",
                "role": "doctor",
                "group": "1"
            },
            {
                "email": "xyz@gmail.com",
                "name": "xyz xyx",
                "role": "nurse",
                "group": "1"
            }
        ],
        "path": [],
        "_id": "66529b1cd689be37e01e78c3",
        "notes": {},
        "scenarioId": "65e45789a86f985e1c5af28f",
        "__v": 0
    }
]
```

Currently, the following code produces:
image

This differs from the expected 2 rows due to how the response body is processed.

**In fact, AC 2 and 3 seem to be misinformed.**

For reference, I would have expected something along the lines of:
```csv
email, name, role, group_number, link
abc@gmail.com, abc aba, doctor, 1, https://vps-dev.wdcc.co.nz/play/65e45789a86f985e1c5af28f 
xyz@gmail.com, xyz xyx, nurse, 1, https://vps-dev.wdcc.co.nz/play/65e45789a86f985e1c5af28f 
```
(Note that the link would be the same for everyone due to how we ended up architecting the `/play/:scenarioId` page)

Ultimately, this is a failure in ticket definition - something to mention in the next retro perhaps!
For this ticket, the end point is great - and it's just minor post processing changes (that _hopefully_ should be quite easy).

**Review by wjin-lee — CHANGES_REQUESTED** [review]
Overall great job!

There seems to have been outdated ACs in the issue. I have left a comment on the issue. For now, please follow the comment I've made as new ACs.

We should also mention this case in the sprint retro.

File: frontend/src/containers/pages/ManageGroups/ManageGroupsPage.jsx (hunk 1 above)
Comment (JordanBlenn) [reply frontend/src/containers/pages/ManageGroups/ManageGroupsPage.jsx:81]: See my comment on the issue

Comment (codecreator127) [reply frontend/src/containers/pages/ManageGroups/ManageGroupsPage.jsx]: Fixed aforementioned issues in this comment

**Review by wjin-lee — APPROVED** [review]
lgtm!

**Pull request merged**